Microsoft.Azure.Cosmos/src/Serializer/CosmosDefaultSystemTextJsonSerializer.cs

```diff
@@ -0,0 +1,128 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos
+{
+    using System;
+    using System.IO;
+    using System.Reflection;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+
+    /// <summary>
+    /// This class provides a default implementation of System.Text.Json Cosmos Linq Serializer.
+    /// </summary>
+    public class CosmosDefaultSystemTextJsonSerializer : CosmosLinqSerializer
+    {
+        /// <summary>
+        /// A read-only instance of <see cref="JsonSerializerOptions"/>.
+        /// </summary>
+        private readonly JsonSerializerOptions jsonSerializerOptions;
+
+        /// <summary>
+        /// Creates an instance of <see cref="CosmosDefaultSystemTextJsonSerializer"/>
+        /// with the default values for the Cosmos SDK
+        /// </summary>
+        /// <param name="jsonSerializerOptions">An instance of <see cref="JsonSerializerOptions"/> containing the json serialization options.</param>
+        public CosmosDefaultSystemTextJsonSerializer(
+            JsonSerializerOptions jsonSerializerOptions)
+        {
+            this.jsonSerializerOptions = jsonSerializerOptions;
+        }
+
+        /// <inheritdoc/>
+        public override T FromStream<T>(Stream stream)
+        {
+            if (stream == null)
+                throw new ArgumentNullException(nameof(stream));
+
+            if (stream.CanSeek && stream.Length == 0)
+            {
+                return default;
+            }
+
+            if (typeof(Stream).IsAssignableFrom(typeof(T)))
+            {
+                return (T)(object)stream;
+            }
+
+            using (stream)
+            {
+                using StreamReader reader = new (stream);
+                return JsonSerializer.Deserialize<T>(reader.ReadToEnd(), this.jsonSerializerOptions);
```

Comment:
```suggestion
                return JsonSerializer.Deserialize<T>(stream, this.jsonSerializerOptions);
```
STJ is able to deserialize the stream directly without needing to allocate an intermediate string.